### PR TITLE
fix: Buy now filter not working

### LINKED ISSUE #6865: Buy now filter not working in explorer
Open explore: Select items and select Buy Now filter: The buy now filter is not working - on production. Tried it on all chains - not working anywhere.

Example RMRK2: You can clearly see that it is showing also NFTs that are not for sale. 
Screenshot 2023-08-21 at 20 40 04

### PR DESCRIPTION
**Pull request by stephenjason89**
## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context
- [x] Closes #6865

- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=16UcV9V6nVvPYdHz98ymUKmNLkzjCEU5sbKJMi7hxYyTHjzR&usdamount=100&donation=true)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 769abca</samp>

Refactored the code for opening the shopping cart modal from the navbar. Moved the logic and configuration from `ShoppingCartModalConfig.ts` and `ShoppingCartButton.vue` to `Navbar.vue` to improve code organization and performance.

### <samp>🤖 Generated by Copilot at 769abca</samp>

> _`ShoppingCartButton`_
> _Simpler, no more `emit` -_
> _Autumn leaves falling_

### CODE REVIEW

**Review by daiagi — CHANGES_REQUESTED** [review]
bug still exists:

image

**Comment by stephenjason89** [comment]
@daiagi Let me check this real quick.

**Comment by stephenjason89** [comment]
I was informed that the breaking change was caused by the recent merges. 
So that was what i fixed.

However, to address the additional concern that you saw.

I corrected the graphql query to only include greater than 0. 
And only include gte when min is > 0. 

Would you want to include the min if it is explicitly set to 0 or just when it's set above 0?

@daiagi please review thank you

File: components/items/ItemsGrid/useItemsGrid.ts [hunk 1]
```diff
@@ -53,33 +53,28 @@ export function useFetchSearch({
...
-        }
+    const variables = search?.length
+      ? { search }
       : {
-          denyList: getDenyList(urlPrefix.value),
-          orderBy: route.query.sort?.length
-            ? route.query.sort
-            : ['blockNumber_DESC'],
           search: searchParams.value,
+          denyList: getDenyList(urlPrefix.value),
```
Comment (roiLeo) [inline components/items/ItemsGrid/useItemsGrid.ts:60]: this should be always in `variables`

**Review by roiLeo — COMMENTED** [review]
otherwise lgtm

File: components/items/ItemsGrid/useItemsGrid.ts (hunk 1 above)
Comment (stephenjason89) [reply components/items/ItemsGrid/useItemsGrid.ts:60]: image
The original code says it should only be present if !search which i corrected to !search?.length

Comment (stephenjason89) [reply components/items/ItemsGrid/useItemsGrid.ts:60]: I usually just stick to the current issue to do my changes. But I'm happy to change it to that. Please confirm @daiagi @Jarsen136 .

**Comment by daiagi** [comment]
> Always return an array thus breaking this in useItemGrid

That was poorly written by me, parsrSearch should return undefined when no search prop is given

Also works roileo is correct and deny query should exist regardless of search prop supplied or not

**Comment by prury** [comment]
what happens here: 

https://github.com/kodadot/nft-gallery/assets/36627808/ed19c5c0-82db-4790-8bb1-e42afbeeaa28

**Comment by stephenjason89** [comment]
@prury Sorry wasn't able to test my last commit. It should be fixed now together with the denyList that wasn't included by @daiagi .

Thanks so much guys

**Comment by stephenjason89** [comment]
> That was poorly written by me

That's alright bro. We all make bugs 🐞 


> parseSearch should return undefined when no search prop is given

Since search prop can be an empty object, it was safer to check the `search?.length` as this would be true on both cases (undefined or empty obj search prop) 

i believe with this, we don't need to change it to return `undefined` but if we do change that will still work. 
_Bonus: it doesn't take up 1 line_


@prury Sorry wasn't able to test my last commit. It should be fixed now together with the denyList that wasn't included. Thanks @roiLeo for pointing it out and @daiagi  for confirming.

I have referenced this https://github.com/kodadot/nft-gallery/pull/6868#pullrequestreview-1589351287 in a separate issue to limit the scope of this PR. already created pr #6894 for it.




Thanks so much guys

**Review by daiagi — APPROVED** [review]

**Review by roiLeo — COMMENTED** [review]
On https://deploy-preview-6868--koda-canary.netlify.app/ksm/explore/items?listed=true&page=2
still show
Capture d’écran 2023-08-23 à 8 53 16 AM

**Comment by stephenjason89** [comment]
> On https://deploy-preview-6868--koda-canary.netlify.app/ksm/explore/items?listed=true&page=2
> 
> still show
> 
> Capture d’écran 2023-08-23 à 8 53 16 AM
> 
> 

That's a different issue #6894 fixes it

**Review by roiLeo — APPROVED** [review]
Now looks good!

**Comment by stephenjason89** [comment]
Thank you @roiLeo ♥️

**Comment by roiLeo** [comment]
- close #6912

**Comment by yangwao** [comment]
pay 70 usd 
Extra effort!

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $70 @ 4.49 USD/DOT ~ 15.59 $DOT
🧗 [16UcV9V6nVvPYdHz98ymUKmNLkzjCEU5sbKJMi7hxYyTHjzR](https://polkadot.subscan.io/account/16UcV9V6nVvPYdHz98ymUKmNLkzjCEU5sbKJMi7hxYyTHjzR)
🔗 [0xf65cd41acf5796ec8f11b6f2c7106d7b569dda67205c351d10b406867783dc58](https://polkadot.subscan.io/extrinsic/0xf65cd41acf5796ec8f11b6f2c7106d7b569dda67205c351d10b406867783dc58)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by stephenjason89** [comment]
Thank you @yangwao ♥️

**Pull request merged**